web/src/features/prompts/server/routers/promptRouter.ts

```diff
@@ -312,12 +313,44 @@ export const promptRouter = createTRPCRouter({
           ctx.prisma,
         );
 
-        await ctx.prisma.prompt.delete({
-          where: {
-            id: input.promptVersionId,
-            projectId: input.projectId,
-          },
-        });
+        const transaction = [
```

Comment:
this seems like an interactive transaction but done without using interactive transactions. I'd assume these prisma calls to be executed before hitting `ctx.prisma.$transaction`. did you check that these actually get processed inside of a transaction?

I'd suggest to make them non-interactive
- fetch next `latest` prompt version outside of transaction
-  delete current latest version and mark next latest as latest within a transaction